Fix Type Annotation in  pandas.compat

Part of #25882

Type annotations for the following file need to be corrected to remove **pandas.compat** from the mypy blacklist. 

Mypy throws the following error:
```sh
pandas/compat/__init__.py:81: error: Module has no attribute "re"
```

PR to follow.

--- a/mypy.ini
+++ b/mypy.ini
@@ -5,9 +5,6 @@ follow_imports=silent
 [mypy-pandas.conftest,pandas.tests.*]
 ignore_errors=True
 
-[mypy-pandas.compat]
-ignore_errors=True
-
 [mypy-pandas.core.api]
 ignore_errors=True
 

--- a/pandas/compat/__init__.py
+++ b/pandas/compat/__init__.py
@@ -11,9 +11,9 @@ Other items:
 * platform checker
 """
 import platform
-import re
 import struct
 import sys
+from typing import Pattern
 
 PY36 = sys.version_info >= (3, 6)
 PY37 = sys.version_info >= (3, 7)
@@ -74,11 +74,7 @@ def raise_with_traceback(exc, traceback=Ellipsis):
     raise exc.with_traceback(traceback)
 
 
-if PY36:
-    from typing import Pattern, Type
-    re_type = Pattern  # type: Type[Pattern]
-else:
-    re_type = type(re.compile(''))
+re_type = Pattern
 
 
 # https://github.com/pandas-dev/pandas/pull/9123

--- a/pandas/compat/__init__.py
+++ b/pandas/compat/__init__.py
@@ -74,11 +74,9 @@ def raise_with_traceback(exc, traceback=Ellipsis):
     raise exc.with_traceback(traceback)
 
 
-# In Python 3.7, the private re._pattern_type is removed.
-# Python 3.5+ have typing.re.Pattern
 if PY36:
-    import typing
-    re_type = typing.re.Pattern
+    from typing import Pattern, Type
+    re_type = Pattern  # type: Type[Pattern]
 else:
     re_type = type(re.compile(''))
 

